attachments/views.py

```diff
@@ -60,8 +69,29 @@ def attach(request, session_id):
                     session.data = {key: value}
             session.save()
             return JsonResponse({'ok': True, 'file_name': f.name, 'file_size': f.size}, content_type=content_type)
+        except FileSizeException as ex:
+            return JsonResponse({'ok': False, 'error': unicode(ex)}, content_type=content_type)
+        except FileTypeException as ex:
+            return JsonResponse({'ok': False, 'error': unicode(ex)}, content_type=content_type)
         except VirusFoundException as ex:
-            logger.exception(str(ex))
+            now = datetime.datetime.now()
+            now = now.strftime('%m-%d-%Y  %H:%M:%S')
+            #request.user may not exist so set up user_prefix to use right prefix for messages on virus upload
+            user = getattr(request, 'user', None)
+            if user:
+                user_prefix = 'User ' + str(user) + ' '
+            else:
+                user_prefix = 'A user '
+            log_message = "attempted to upload this file: %s with virus signature: %s at %s" % (f.name, virus[path][1],now)
+            log_message = user_prefix + log_message
+            logger.exception(log_message)
+            #if ATTACHMENTS_VIRUS_EMAIL is set to a list/tuple of email addresses to send to it will send email alert
+            if getattr(settings, 'ATTACHMENTS_VIRUS_EMAIL', False):
+                #send email to email list
+                email_list = getattr(settings, 'ATTACHMENTS_VIRUS_EMAIL')
+                subject = 'VIRUS UPLOAD ALERT: " + user_prefix + "attempted to upload a file containing a virus to the system'
+                message = log_message
```

Comment:
Instead of building this message in code, why not add a new `templates/attachments/virus_email.txt` template, and use `loader.render_to_string` to build the message. Has the advantage that sites can provide their own templates if they want.